packages/mdc-dialog/mdc-dialog.scss

```diff
@@ -26,147 +26,203 @@
 @import "@material/rtl/mixins";
 @import "@material/typography/mixins";
 @import "@material/typography/variables";
+@import "./mixins";
+@import "./variables";
 
 // postcss-bem-linter: define dialog
 
 .mdc-dialog {
-  display: flex;
+  // Use `display: none` instead of `visibility: hidden` to work around a `-webkit-line-clamp` bug.
+  // https://stackoverflow.com/a/44193578/467582
+  display: none;
+
+  // stylelint-disable-next-line declaration-empty-line-before
   position: fixed;
   top: 0;
   left: 0;
   align-items: center;
   justify-content: center;
+  box-sizing: border-box;
   width: 100%;
   height: 100%;
-  visibility: hidden;
   z-index: 5;
 }
 
-.mdc-dialog__backdrop {
+.mdc-dialog__scrim {
   @include mdc-theme-prop(background-color, text-primary-on-light);
 
   position: fixed;
   top: 0;
   left: 0;
   align-items: center;
   justify-content: center;
+  box-sizing: border-box;
   width: 100%;
   height: 100%;
   opacity: 0;
   z-index: -1;
 }
 
-.mdc-dialog__surface {
+.mdc-dialog__container {
+  @include mdc-dialog-corner-radius($mdc-dialog-corner-radius);
   @include mdc-elevation(24);
   @include mdc-theme-prop(background-color, background);
 
   display: inline-flex;
   flex-direction: column;
-  width: calc(100% - 30px);
-  min-width: 640px;
-  max-width: 865px;
+  box-sizing: border-box;
+  min-width: $mdc-dialog-min-width;
+  max-height: $mdc-dialog-viewport-size-expr;
   transform: translateY(150px) scale(.8);
-  border-radius: 2px;
   opacity: 0;
 
+  // Fit snugly within the viewport at smaller screen sizes.
+  @media (max-width: ($mdc-dialog-width-breakpoint - 1)) {
+    max-width: $mdc-dialog-viewport-size-expr;
+  }
+
+  // Once the screen gets big enough, apply a fixed maximum width.
```

Comment:
It's to ensure that prose line lengths are readable.

Otherwise, at larger screen sizes you get this (super hard to read):

![image](https://user-images.githubusercontent.com/409245/44803499-7e19aa00-ab73-11e8-8273-da59c3d4926b.png)

I was planning to add mixins to let clients override the default min/max width and height.